lib/Basic/FileSystem.cpp

```diff
@@ -30,6 +36,136 @@ namespace {
   };
 } // end anonymous namespace
 
+/// Does some simple checking to see if a temporary file can be written next to
+/// \p outputPath and then renamed into place.
+///
+/// Helper for swift::atomicallyWritingToFile.
+///
+/// If the result is an error, the write won't succeed at all, and the calling
+/// operation should bail out early.
+static llvm::ErrorOr<bool>
+canUseTemporaryForWrite(const StringRef outputPath) {
+  namespace fs = llvm::sys::fs;
+
+  if (outputPath == "-") {
+    // Special case: "-" represents stdout, and LLVM's output stream APIs are
+    // aware of this. It doesn't make sense to use a temporary in this case.
+    return false;
+  }
+
+  fs::file_status status;
+  (void)fs::status(outputPath, status);
+  if (fs::exists(status)) {
```

Comment:
Probably reads slightly better if you flip the "if" and use an early return.

Reply:
Oh, of course. This is left over from before I pulled it out into its own function. Easy enough.